packages/dds/tree/src/test/tree/anchorSet.spec.ts

```diff
@@ -315,14 +320,54 @@ describe("AnchorSet", () => {
 			// This moves anchor4 (the only anchor under anchor1) out from under anchor1.
 			// If the visitor did not increase the ref count of anchor1 on its way down,
 			// anchor1 will be disposed as part of this operation.
-			v.detach({ start: 4, end: 5 }, detachedField);
+			v.detach({ start: 4, end: 5 }, detachedField, detachId);
 			v.exitField(fieldBar);
 			// If anchor1 is be disposed. This will throw.
 			v.exitNode(5);
 			v.exitField(fieldFoo);
 		});
 
-		checkRemoved(anchors.locate(anchor1), detachedField);
+		checkRemoved(anchors.locate(anchor1), detachId, detachedField);
+	});
+
+	it("can detach multiple nodes", () => {
+		const anchors = new AnchorSet();
+
+		// This tests that detaching nodes [a, b, c, d, e] while there are anchors to b and e works
+		const bPath = makePath([fieldFoo, 1]);
+		const ePath = makePath([fieldFoo, 4]);
+		const bAnchor = anchors.track(bPath);
+		const eAnchor = anchors.track(ePath);
+
+		withVisitor(anchors, (v) => {
+			v.enterField(fieldFoo);
+			v.detach({ start: 0, end: 5 }, detachedField, detachId);
+			v.exitField(fieldFoo);
+		});
+
+		checkRemoved(anchors.locate(bAnchor), { minor: 43 }, detachedField);
+		checkRemoved(anchors.locate(eAnchor), { minor: 46 }, detachedField);
+	});
+
+	it("does not retain detachedNodeIds when detached nodes are reattached", () => {
+		const anchors = new AnchorSet();
+
+		// This tests that detaching nodes [a, b, c, d, e] while there are anchors to b and e works
+		// and that reattaching [a, b, c] while there is an anchor to b removes the detachedNodeId from b
+		const bPath = makePath([fieldFoo, 1]);
+		const ePath = makePath([fieldFoo, 4]);
+		const bAnchor = anchors.track(bPath);
+		const eAnchor = anchors.track(ePath);
+
+		withVisitor(anchors, (v) => {
+			v.enterField(fieldFoo);
+			v.detach({ start: 0, end: 5 }, detachedField, detachId);
+			v.attach(detachedField, 3, 0);
+			v.exitField(fieldFoo);
+		});
+
+		assert(isDetachedUpPath(anchors.locate(bAnchor) as UpPath) === false);
+		assert(isDetachedUpPath(anchors.locate(eAnchor) as UpPath) === true);
```

Comment:
Nit: it's safest to go for the most gentle cast possible:
```suggestion
		assert(isDetachedUpPath(anchors.locate(bAnchor)!) === false);
		assert(isDetachedUpPath(anchors.locate(eAnchor)!) === true);
```
or not cast at all:
```suggestion
		const bPathLookup = anchors.locate(bAnchor);
		const ePathLookup = anchors.locate(eAnchor);
		assert(bPathLookup !== undefined);
		assert(ePathLookup !== undefined);
		assert(isDetachedUpPath(bPathLookup) === false);
		assert(isDetachedUpPath(ePathLookup) === true);
```